circuit/src/gadgets/range.rs

```diff
@@ -0,0 +1,476 @@
+							.collect::<Vec<_>>()
+					})
+					.transpose_vec(N);
+
+				//
+				// x:   1111111100000001
+				// a_0: 11111111
+				// a_1: 00000001
+				//
+				// z_0 = x
+				// z_1 = (z_0 - a_0) / (2 ^ K)
+				// z_2 = (z_1 - a_1) / (2 ^ K)
+				//
+				// [1111111100000001, 00000001, 0]
+				//
+
+				// Assign cumulative sum such that
+				//          z_i = 2^{K}⋅z_{i + 1} + a_i
+				// => z_{i + 1} = (z_i - a_i) / (2^K)
+				//
+				// For `element` = a_0 + 2^10 a_1 + ... + 2^{120} a_{12}}, initialize z_0 =
+				// `element`. If `element` fits in 130 bits, we end up with z_{13} = 0.
+				let mut z = x.clone();
+				let mut last_word_cell = x.clone();
+				let words_len = words.len();
+				let inv_two_pow_k = F::from(1u64 << K).invert().unwrap();
+				for (id, word) in words.into_iter().enumerate() {
+					// Enable q_lookup on this row
+					ctx.enable(selector.clone())?;
+
+					if id == words_len - 1 {
+						last_word_cell = z.clone();
+					}
+
+					// z_next = (z_cur - m_cur) / 2^K
+					let z_next = z.value().zip(word).map(|(z, word)| (*z - word) * inv_two_pow_k);
+
+					// Assign z_next
+					ctx.next();
+					z = ctx.assign_advice(mock_common.common.advice[0], z_next)?;
+				}
+
+				ctx.constrain_to_constant(z.clone(), F::zero())?;
+
+				Ok((self.x.clone(), last_word_cell))
+			},
+		)
+	}
+}
+
+/// RangeChipsetConfig
+#[derive(Debug, Clone)]
+pub struct RangeChipsetConfig {
+	running_sum_selector: Selector,
+	short_word_selecor: Selector,
+}
+
+impl RangeChipsetConfig {
+	/// Construct a new instance
+	pub fn new(running_sum_selector: Selector, short_word_selecor: Selector) -> Self {
+		Self { running_sum_selector, short_word_selecor }
+	}
+}
+
+/// RangeChipset
+#[derive(Debug, Clone)]
+pub struct RangeChipset<F: FieldExt, const K: usize, const N: usize, const S: usize> {
+	x: AssignedCell<F, F>,
+}
+
+impl<F: FieldExt, const K: usize, const N: usize, const S: usize> RangeChipset<F, K, N, S> {
+	/// Constructs new chipset
+	pub fn new(x: AssignedCell<F, F>) -> Self {
+		Self { x }
+	}
+}
+
+impl<F: FieldExt, const K: usize, const N: usize, const S: usize> MockChipset<F>
+	for RangeChipset<F, K, N, S>
+{
+	type Config = RangeChipsetConfig;
+	type Output = AssignedCell<F, F>;
+
+	fn synthesize(
+		self, mock_common: &MockCommonConfig, config: &RangeChipsetConfig,
+		mut layouter: impl halo2::circuit::Layouter<F>,
+	) -> Result<Self::Output, Error> {
+		// First, check if x is less than 256 bits
+		let range_chip = LookupRangeCheckChip::<F, K, N>::new(self.x.clone());
+		let (x, last_word) = range_chip.synthesize(
+			mock_common,
+			&config.running_sum_selector,
+			layouter.namespace(|| "range check"),
+		)?;
+
+		let last_word_cell = layouter.assign_region(
+			|| "copy last word of x",
+			|region| {
+				let mut ctx = RegionCtx::new(region, 0);
+				ctx.copy_assign(mock_common.common.advice[0], last_word.clone())
+			},
+		)?;
+		let short_word_chip = LookupShortWordCheckChip::<F, K, S>::new(last_word_cell);
```

Comment:
We can just pass `last_word` into the `LookupShortWordCheckChip` constructor.
The value will be copy assigned inside it

Reply:
Done https://github.com/eigen-trust/protocol/pull/169/commits/2b7cbfb7f9b4e8032e5c349a1e5f86dd98194146